libc-top-half/musl/src/unistd/getcwd.c

```diff
@@ -4,6 +4,8 @@
 #include <string.h>
 #include "syscall.h"
 
+char __wasilibc_cwd[PATH_MAX] = {'/', 0};
```

Comment:
WASI libc may move to deprecate `PATH_MAX` since every OS has its own limit, and it's desirable for WASI libc to avoid baking in a particular limit. We can use a `malloc` buffer here to avoid having a fixed size limit, with static initialization like this:

```c
char __wasilibc_cwd_buffer[] = {'/', 0};
char __wasilibc_cwd = __wasilibc_cwd_buffer;
```

This will require checking for `__wasilibc_cwd == __wasilibc_cwd_buffer` to avoid doing `free` or `realloc` on the static buffer.